server/pbenchinacan/deploy

```diff
@@ -1,26 +1,84 @@
 #!/bin/bash -e
 
-# Update the dashboard code
-podman unshare rm -r /srv/pbench/public_html/dashboard/
-podman unshare cp -r ${HOME}/git/pbench/dashboard/build /srv/pbench/public_html/dashboard
+#
+# This script deploys a Pbench Server instance from a container image.
+#
+# Steps performed:
+#   - The Pbench Dashboard application files are copied from the build directory
+#     into the SRV_PBENCH file system.  (If the build directory does not exist,
+#     an empty directory is created at the input and output locations, a warning
+#     is printed, and the Pbench Server is deployed without a Dashboard.)
+#   - If present in the current directory, the Nginx favicon file is copied into
+#     the appropriate location in the SRV_PBENCH file system.
+#   - The ownership of the files in the SRV_PBENCH file system will are set for
+#     access inside the container.
+#   - The Pbench Server container is started and run as a detached process.
+#
+# The key inputs are the following environment variables:
+#   - PB_DEPLOY_FILES:  the location (on the host) of the configuration files
+#                       for the Pbench Server and rsyslog.
+#   - SRV_PBENCH:  the location of the file system (on the host) which will be
+#                       mounted inside the container at /srv/pbench.
+#
+# Other inputs which can be provided by environment variables include:
+#   - the name and pull policy for the Pbench Server container image
+#   - the location of the Pbench Dashboard app files and the Nginx Favicon
+#   - the location of the Pbench Git checkout (used to locate the above)
+#
+# In all cases, suitable defaults are provided for any value which is undefined.
+#
 
-# Update the favicon
-podman unshare rm /srv/pbench/public_html/favicon.ico
-podman unshare cp color-square.ico /srv/pbench/public_html/favicon.ico
+#+
+# Configuration
+#-
 
-IMAGE=images.paas.redhat.com/pbench/pbench-server:main
+# Pbench Server container image
+PB_SERVER_IMAGE=${PB_SERVER_IMAGE:-"images.paas.redhat.com/pbench/pbench-server:main"}
+PB_SERVER_IMAGE_PULL_POLICY=${PB_SERVER_IMAGE_PULL_POLICY:-"newer"}
 
-# First ensure /srv/pbench is owned by the "pbench" user INSIDE the container we run.
-podman run --rm --volume /srv/pbench:/srv/pbench:Z ${IMAGE} chown -R pbench:pbench /srv/pbench
+# Locations on the host
+PB_DASHBOARD_DIR=${PB_DASHBOARD_DIR:-${GITTOP:=$(git rev-parse --show-toplevel)}/dashboard/build}
+NGINX_FAVICON=${NGINX_FAVICON:-${GITTOP:=$(git rev-parse --show-toplevel)}/server/lib/config/color-square.ico}
```

Comment:
Since I imagine you'll ask, @dbutenhof, the reference to `GITTOP` will _assign_ the variable if it is undefined as well as evaluate it.  I needed to specify the full reference in both lines because, if `PB_DASHBOARD_DIR` is defined, then the assignment won't happen in the first line.

I purposefully did not put `GITTOP` on it's own line:  if the caller defines both `PB_DASHBOARD_DIR` and `NGINX_FAVICON`, then we won't invoke Git and we won't even need to have it installed, which I thought made the replication worthwhile.